src/lager_handler_watcher.erl

```diff
@@ -194,5 +192,17 @@ reinstall_on_runtime_failure_test_() ->
         ]
     }.
 
+pop_until(String, Fun) ->
```

Comment:
Yes, that was the question. `lists:flatten` makes a lot of sense.  The others might depend on use case. Thanks.